db.go

```diff
@@ -292,9 +295,13 @@ func Open(dir string, l log.Logger, r prometheus.Registerer, opts *Options) (db
 	if opts.WALSegmentSize > 0 {
 		segmentSize = opts.WALSegmentSize
 	}
-	wlog, err := wal.NewSize(l, r, filepath.Join(dir, "wal"), segmentSize)
-	if err != nil {
-		return nil, err
+	var wlog *wal.WAL
+	if opts.WALSegmentSize >= 0 {
```

Comment:
```
	// Wal is enabled.
	if opts.WALSegmentSize >= 0 {
		// Wal is set to a custom size.
		if opts.WALSegmentSize > 0 {
			segmentSize = opts.WALSegmentSize
		}	
		wlog, err = wal.NewSize(l, r, filepath.Join(dir, "wal"), segmentSize)
		if err != nil {
			return nil, err
		}	
	}
```

Reply:
I'm so sorry. I somehow did not see this. I have updated.